scenarios/kubernetes_e2e.py

```diff
@@ -181,6 +188,8 @@ def main(args):
         '--cluster', default='bootstrap-e2e', help='Name of the cluster')
     PARSER.add_argument(
         '--tag', default='v20170104-9031f1d', help='Use a specific kubekins-e2e tag if set')
+    PARSER.add_argument(
+        '--docker-in-docker', help='Enable run docker within docker')
```

Comment:
Have you tested this? 

I'm pretty sure you want to add `action='store_true'` and that it will fail as written.

Also consider adding a TODO to change the other boolean arguments to follow this pattern (for a separate PR):

```
def silly(b):
  return 'true' if b else 'false'

def foo(up):
  cmd.extend(['-e', 'E2E_UP=%s' % silly(up)])
```